[`MINOR`] Fix buggy CI self hosted runners behaviour `flash failed`

Sometimes it's needed to re-run a hardware CI job multiple times for it to succeed. It seems to be caused by the runner not being able to properly access the ST-Link. A few examples of false fails:

```bash
[100%] Built target OBC.elf
st-flash 1.6.1
2022-08-21T14:33:55 INFO common.c: F4xx (Dynamic Efficency): 96 KiB SRAM, 384 KiB flash in at least 16 KiB pages.
file OBC.bin md5 checksum: 22fb569ed5f2276ebadbfe39ed765e, stlink checksum: 0x0008c5b6
2022-08-21T14:33:55 INFO common.c: Attempting to write 6116 (0x17e4) bytes to stm32 address: 134217728 (0x8000000)
EraseFlash - Sector:0x0 Size:0x4000 2022-08-21T14:33:55 INFO common.c: Flash page at addr: 0x08000000 erased
2022-08-21T14:33:55 INFO common.c: Finished erasing 1 pages of 1[63](https://github.com/Aerospace-ASP/Firmware/runs/7938859229?check_suite_focus=true#step:6:64)84 (0x4000) bytes
2022-08-21T14:33:55 INFO common.c: Starting Flash write for F2/F4/L4
2022-08-21T14:33:55 INFO flash_loader.c: Successfully loaded flash loader in sram
enabling 32-bit flash writes
size: 6116
2022-08-21T14:33:56 ERROR flash_loader.c: flash loader run error
stlink_fwrite_flash() == -1
2022-08-21T14:33:56 ERROR common.c: stlink_flash_loader_run(0x8000000) failed! == -1
Error: Process completed with exit code 255.
```

```bash
Building: OBC.bin
[100%] Built target OBC.elf
libusb: error [get_usbfs_fd] libusb couldn't open USB device /dev/bus/usb/001/003, errno=13
st-flash 1.6.1
libusb: error [get_usbfs_fd] libusb requires write access to USB device nodes
2022-08-20T17:46:01 WARN usb.c: Couldn't find any ST-Link devices
Error: Process completed with exit code 255.
```

Examples:
* https://github.com/Aerospace-ASP/Firmware/runs/7932621014?check_suite_focus=true
* https://github.com/Aerospace-ASP/Firmware/runs/7715196487?check_suite_focus=true
* https://github.com/Aerospace-ASP/Firmware/runs/7715190035?check_suite_focus=true
* https://github.com/Aerospace-ASP/Firmware/runs/7939927833?check_suite_focus=true

Related to:
* https://github.com/Aerospace-ASP/Firmware/issues/5

--- a/.github/workflows/hardware.yml
+++ b/.github/workflows/hardware.yml
@@ -16,5 +16,6 @@ jobs:
             - run: cmake --version || sudo apt install -y cmake
             - run: arm-none-eabi-gcc --version || sudo apt install -y gcc-arm-none-eabi
             # Build the project and flash it
+            - run: sleep 1 # add a sleep command so that we wait out the blocking of the device
             - run: mkdir build && cd build && cmake -DCMAKE_BUILD_TYPE=Debug -DCMAKE_TOOLCHAIN_FILE=../cmake/arm-none-eabi-gcc-toolchain.cmake .. && make -j`nproc` && st-flash write OBC.bin 0x8000000
 

--- a/.github/workflows/hardware.yml
+++ b/.github/workflows/hardware.yml
@@ -17,5 +17,5 @@ jobs:
             - run: arm-none-eabi-gcc --version || sudo apt install -y gcc-arm-none-eabi
             # Build the project and flash it
             - run: sleep 1 # add a sleep command so that we wait out the blocking of the device
-            - run: mkdir build && cd build && cmake -DCMAKE_BUILD_TYPE=Debug -DCMAKE_TOOLCHAIN_FILE=../cmake/arm-none-eabi-gcc-toolchain.cmake .. && make -j`nproc` && st-flash write OBC.bin 0x8000000
+            - run: mkdir build && cd build && cmake -DCMAKE_BUILD_TYPE=Debug -DCMAKE_TOOLCHAIN_FILE=../cmake/arm-none-eabi-gcc-toolchain.cmake .. && make -j`nproc` && st-flash write OBC.bin 0x8000000 || st-flash write OBC.bin 0x8000000
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -12,3 +12,7 @@
 - Added CLion as a supported `IDE`. ([#10](https://github.com/Aerospace-ASP/Firmware/pull/10))
 - Added Dockerfiles for the projects environment to Docker Hub. ([#10](https://github.com/Aerospace-ASP/Firmware/pull/10))
 - Added a basic CI configuration with self hosted runners for testing. ([#10](https://github.com/Aerospace-ASP/Firmware/pull/10))
+
+### Fixed
+
+- Added a temporary fix for the flashing failure in the hardware CI job by retrying it once and adding 1 second delay. ([#36](https://github.com/Aerospace-ASP/Firmware/issues/36))

